compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -894,6 +894,14 @@ public void visit(BLangImportPackage importPkgNode) {
 
         PackageID pkgId = new PackageID(orgName, nameComps, version);
 
+        // Un-exported modules is not allowed to import.
+        BPackageSymbol bPackageSymbol = this.packageCache.getSymbol(pkgId);
+        if (bPackageSymbol != null && bPackageSymbol.moduleContext() != null
```

Comment:
Will this also forbid importing an unexported module within the same package? Importing within the same package should be allowed.

Reply:
Shall we add another test case to check the behavior when an unexported module is imported within the same package?

Reply:
Added